features/FxA-45-device-registration-api/FX-DESKTOP.md

```diff
@@ -0,0 +1,122 @@
+Firefox Device Registration
+===========================
+
+This is a breakdown of the states Firefox needs to handle/worry about.
```

Comment:
It would also be useful to add a description of what the desired "steady state" is, i.e. that in the absence of any changes a connected device should:

* have a deviceID and a sessionToken
* use its deviceID to identify itself in the sync clients collection
* have a name that's consistent in local storage, fxa server record, and sync clients record